server/controllers/shoppingList.js

```diff
@@ -196,35 +197,47 @@ const addProductToList = (req, resp) => {
 };
 
 const getListData = (req, resp) => {
-  ShoppingList.find(
-    {
-      _id: req.params.id,
-      $or: [
-        { adminIds: req.user._id },
-        { ordererIds: req.user._id },
-        { purchaserIds: req.user._id }
-      ]
-    },
-    (err, documents) => {
-      if (err) {
-        return resp.status(404).send({ message: err.message });
+  let list;
+  ShoppingList.findOne({
+    _id: req.params.id,
+    $or: [
+      { adminIds: req.user._id },
+      { ordererIds: req.user._id },
+      { purchaserIds: req.user._id }
+    ]
+  })
+    .then(doc => {
+      if (!doc) {
+        return resp.status(404).send({ message: 'No list of given id!' });
       }
 
-      if (documents && documents.length > 0) {
-        const { cohortId, _id, isArchived } = documents[0];
+      list = doc;
```

Comment:
maybe try to return doc at the end of then and check if you have access to this in next then

Reply:
Checking cohort isArchive status is async task, so the doc can't be returned because next 'then' won't wait for cohort check...